src/lib/Libifl/pbsD_selectj.c

```diff
@@ -58,9 +58,118 @@
 #include "dis.h"
 #include "pbs_ecl.h"
 
+struct reply_list {
+	struct batch_reply *reply;
+	struct reply_list *next;
+	struct reply_list *last;
+};
+
+/**
+ * @brief
+ *	-reads selectjob reply from stream
+ *
+ * @param[in] c - communication handle
+ * @param[in] jlist - structure which holds job list and related counters.
+ *
+ * @return	string list
+ * @retval	list of strings		success
+ * @retval	NULL			error
+ *
+ */
+static void
+PBSD_select_get(int c, struct reply_list **rlist)
+{
+	struct batch_reply *reply;
+
+	/* read reply from stream */
+
+	reply = PBSD_rdrpy(c);
+	if (reply == NULL) {
+		pbs_errno = PBSE_PROTOCOL;
+	} else if (reply->brp_choice != BATCH_REPLY_CHOICE_NULL &&
+		   reply->brp_choice != BATCH_REPLY_CHOICE_Text &&
+		   reply->brp_choice != BATCH_REPLY_CHOICE_Select) {
+		pbs_errno = PBSE_PROTOCOL;
+	} else if (get_conn_errno(c) == 0) {
+		if (reply->brp_un.brp_select == NULL) {
+			PBSD_FreeReply(reply);
+			return;
+		}
+
+		struct reply_list *new = malloc(sizeof(struct reply_list));
+		new->reply = reply;
+		new->next = NULL;
+		new->last = new;
+		if (*rlist) {
+			(*rlist)->last->next = new;
+			(*rlist)->last = new;
+		} else
+			*rlist = new;
+	}
+}
+
+/**
+ * @brief
+ *	-convert the reply list to array of jobids
+ *
+ * @param[in] rlist - reply list
+ *
+ * @return	string list
```

Comment:
This list you are returning has data that points to location within the array. Please write a comment about format of this list.